### ISSUE TITLE
DirectoryTree component not working when expandedKeys=undefined

### ISSUE THREAD

**Issue by SquirrelDeveloper** [ant-design/ant-design#49668]
### Reproduction link

[https://antd-reproduce-5x-7veb3c.stackblitz.io](https://antd-reproduce-5x-7veb3c.stackblitz.io)

### Steps to reproduce

When, in Tree (DirectoryTree) component the prop expandedKeys is specified and set to either undefined or [], the tree nodes are not expandable anymore

### What is expected?

When expandedKeys is set to undefined, you should still be able to expand any node you like

### What is actually happening?

The nodes are not expandable

<img width="1362" alt="Screenshot 2024-07-01 at 11 16 16" src="https://github.com/ant-design/ant-design/assets/172485993/b11003df-24dc-4041-bb21-381c3d421163">

| Environment | Info |
| --- | --- |
| antd | 5.18.3 |
| React | 18.3.1 |
| System | Macos - latest |
| Browser | Chrome - latest |

**Comment by zombieJ** [comment]
ref https://github.com/react-component/tree/blob/master/src/Tree.tsx#L353

`rc-tree` use `in` for check prop. Seems could be improved

**Comment by aojunhao123** [comment]
Hi, I would like to take on this issue and work on a fix. Please assign it to me. Thanks!

**Comment by aojunhao123** [comment]
emmm......果然这个CI还是有用的（

**Comment by giaBaoJS** [comment]
Opened #59076 for the `DirectoryTree` side of this.

On @zombieJ's `rc-tree` pointer — I measured it, and it is a separate bug that this PR does not close: `<Tree expandedKeys={undefined}>` is unexpandable too, while `<Tree>` with the prop omitted is fine. That one needs the `hasOwnProperty` check in `rc-tree`'s `setExpandedKeys` to become a `!== undefined` check.

`DirectoryTree` still needs its own fix regardless, because it keeps a second copy of the check over its own state. The `undefined` it stores there also makes shift-click range selection throw `TypeError: Cannot read properties of undefined (reading 'includes')` in `components/tree/utils/dictUtil.ts`, which no change in `rc-tree` reaches.

**Comment by aojunhao123** [comment]
Indeed, antd clearly defines this behavior, and this is an issue that needs to be fixed.

- https://ant.design/docs/react/faq#is-there-a-difference-between-undefined-and-null-in-the-controlled-components-of-antd
